rfcs/text/semantic-release.md

```diff
@@ -0,0 +1,257 @@
+> 1. **BREAKING CHANGE:** a commit that has a footer `BREAKING CHANGE:`, or appends a `!` after the type/scope, introduces a breaking API change (correlating with [`MAJOR`](https://semver.org/) in semantic versioning). A BREAKING CHANGE can be part of commits of any *type*.
+> 1. *types* other than `fix:` and `feat:` are allowed, for example [`@commitlint/config-conventional`](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional) (based on the the [Angular convention](https://github.com/angular/angular/blob/22b96b9/CONTRIBUTING.md#-commit-message-guidelines)) recommends `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others.
+> 1. footers other than `BREAKING CHANGE: <description> `may be provided and follow a convention similar to [git trailer format](https://git-scm.com/docs/git-interpret-trailers).
+>
+> Additional types are not mandated by the conventional commits specification, and have no implicit effect in semantic versioning (unless they include a BREAKING CHANGE).
+>
+> A scope may be provided to a commit’s type, to provide additional contextual information and is contained within parenthesis, e.g., `feat(parser): add ability to parse arrays`.  
+> [[Source]](https://www.conventionalcommits.org/en/v1.0.0/)
+
+> ## Subject
+> The subject contains a succinct description of the change:
+>
+> * use the imperative, present tense: "change" not "changed" nor "changes"
+> * don't capitalize the first letter
+> * no dot (.) at the end
+> ## Body
+> Just as in the *subject*, use the imperative, present tense: "change" not "changed" nor "changes". The body should include the motivation for the change and contrast this with previous behavior.
+>
+> ## Footer
+> The footer should contain any information about **Breaking Changes** and is also the place to reference GitHub issues that this commit **Closes**.
+>
+> **Breaking Changes** should start with the word `BREAKING CHANGE:` with a space or two newlines. The rest of the commit message is then used for this.  
+> [[Source]](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit)
+
+We will use [`commitlint`](https://github.com/conventional-changelog/commitlint) and [`husky`](https://github.com/typicode/husky) to check our commits against the `conventionalcommits` guidelines.
+- [`husky`](https://github.com/typicode/husky) will manage the git commit hooks defined in `package.json`
+- we will write a hook that runs the commit message via [`commitlint`](https://github.com/conventional-changelog/commitlint)
+- we will use the [`@commitlint/config-conventional`](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional) config
+- [`husky`](https://github.com/typicode/husky) will abort the commit if the commit message is in the wrong format
```

Comment:
If the user has spent some time drafting a detailed commit message, is there a risk that their drafted message could be lost if `husky` aborts the commit if the user made a mistake in the formatting?

Reply:
When using `commitizen` the user can use `npm run commit -- --retry` to re-use the last commit message. I'll mention this in the RFC.

Reply:
Addressed in 00cd9bcc3d25b212ab84801642bbb80f3bf085be